core/metadata/db.go

```diff
@@ -270,6 +270,20 @@ func (m *DB) Update(fn func(*bolt.Tx) error) error {
 	return err
 }
 
+// Publisher returns an event publisher if one is configured
+// and the current context is not inside a transaction.
+func (m *DB) Publisher(ctx context.Context) events.Publisher {
+	_, ok := ctx.Value(transactionKey{}).(*bolt.Tx)
+	if ok {
+		// Do no publish events within a transaction
```

Comment:
This might be part of a normal flow, thats why I put "If we want to support this pattern better in the future we can queue up events during a transaction and publish them on success at the end." If we do that, we could add a publisher to the transaction context which is able to queue up and publish at the end. Right now if this pattern is used (I don't see anywhere that it is), the publishing needs to be handled by caller.